src/sth_database.js

```diff
@@ -236,6 +236,227 @@
+      function (err) {
+        // Once the aggregated data collection has been prepopulated (if needed),
+        //  aggregate the new value received.
+        getAggregatedModel(collectionName4Aggregated).update(
+          getAggregateUpdateCondition(resolution, timestamp),
+          getAggregateUpdate4Update(attributeValue, resolution, timestamp),
+          function (err) {
+            if (callback) {
+              callback(err);
+            }
+          }
+        );
+      }
+    );
+  }
+
+  /**
+   * Stores the aggregated data for a new event (attribute value)
+   * @param {Date} timestamp The date the event arrived
+   * @param {string} entityId The entity id associated to updated attribute
+   * @param {string} entityType The entity type associated to the updated attribute
+   * @param {string} attributeId The updated attribute id
+   * @param {string} attributeType The updated attribute type
+   * @param {string} attributeValue The updated attribute value
+   * @param {Function} callback Function to call once the operation completes
+   */
+  function storeAggregatedData(
+    timestamp, entityId, entityType, attributeId, attributeType, attributeValue, callback) {
+    var collectionName4Aggregated = getCollectionName4Aggregated(entityId, attributeId);
+
+    storeAggregatedData4Resolution(collectionName4Aggregated, sthConfig.RESOLUTION.SECOND, timestamp,
+      entityId, entityType, attributeId, attributeType, attributeValue, callback);
+
+    storeAggregatedData4Resolution(collectionName4Aggregated, sthConfig.RESOLUTION.MINUTE, timestamp,
+      entityId, entityType, attributeId, attributeType, attributeValue, callback);
+
+    storeAggregatedData4Resolution(collectionName4Aggregated, sthConfig.RESOLUTION.HOUR, timestamp,
+      entityId, entityType, attributeId, attributeType, attributeValue, callback);
+
+    storeAggregatedData4Resolution(collectionName4Aggregated, sthConfig.RESOLUTION.DAY, timestamp,
+      entityId, entityType, attributeId, attributeType, attributeValue, callback);
+
+    storeAggregatedData4Resolution(collectionName4Aggregated, sthConfig.RESOLUTION.MONTH, timestamp,
+      entityId, entityType, attributeId, attributeType, attributeValue, callback);
+  }
+
+  /**
+   * Stores the raw data for a new event (attribute value)
+   * @param {Date} timestamp The date the event arrived
+   * @param {string} entityId The entity id associated to updated attribute
+   * @param {string} entityType The entity type associated to the updated attribute
+   * @param {string} attributeId The updated attribute id
+   * @param {string} attributeType The updated attribute type
+   * @param {string} attributeValue The updated attribute value
+   * @param {Function} callback Function to call once the operation completes
+   */
+  function storeRawData(
+    timestamp, entityId, entityType, attributeId, attributeType, attributeValue, callback) {
+    var collectionName4Events = getCollectionName4Events(entityId, attributeId);
```

Comment:
This seems to be creating a collection name based on the entityId and the attributeId... this should be aligned (later or now, I don't know) with the naming conventions we are discussing offline.